src/main/java/org/jabref/gui/dialogs/AutosaveUIManager.java

```diff
@@ -25,10 +29,20 @@ public AutosaveUIManager(BasePanel panel) {
 
     @Subscribe
     public void listen(@SuppressWarnings("unused") AutosaveEvent event) {
-        try {
-            new SaveDatabaseAction(panel, Globals.prefs, Globals.entryTypesManager).save(SaveDatabaseAction.SaveDatabaseMode.SILENT);
-        } catch (Throwable e) {
-            LOGGER.error("Problem occured while saving.", e);
+        if (timer != null) {
+            timer.cancel();
         }
+
+        timer = new Timer();
+        timer.schedule(new TimerTask() {
+            @Override
+            public void run() {
+                try {
+                    new SaveDatabaseAction(panel, Globals.prefs, Globals.entryTypesManager).save(SaveDatabaseAction.SaveDatabaseMode.SILENT);
```

Comment:
Should this be called in the application main thread as the class is called `...UIManager`?

Reply:
I would assume, it is OK as is.